sklearn/decomposition/_kernel_pca.py

```diff
@@ -210,6 +264,28 @@ def _fit_transform(self, K):
                                                 tol=self.tol,
                                                 maxiter=self.max_iter,
                                                 v0=v0)
+        elif eigen_solver == 'randomized':
+            # not using flip_sign=True for deterministic output, we do it later
+            U, S, V = randomized_svd(K, n_components=n_components,
+                                     n_iter=self.iterated_power,
+                                     flip_sign=False,
+                                     random_state=self.random_state)
+            self.alphas_ = U[:, :n_components]  # eigenvectors
+            self.lambdas_ = S[:n_components]    # eigenvalues
+
+            # Conversion of Singular values into Eigenvalues:
+            # Above we use SVD to get the eigenvalues decomposition.
+            # However SVD does not guarantee that sign of u and v is the same.
+            # If they are different, the corresponding *singular value* is not
+            # equal to the *eigenvalue*, it has the wrong sign. We have to
+            # take this into account to find the actual *eigenvalue*
+            VU = np.dot(V[:n_components, :], U[:, :n_components])
```

Comment:
I need to think about this more, but any chance we can avoid a matrix multiplication just to know the signs?

Reply:
There is surely a better way, indeed ! I'll try to think about it again. But this is extremely minor in terms of perfs as compared with the SVD operation :)

Reply:
Maybe we could try to add the np.sign() of each matrix and then when this is zero we need to flip the corresponding singular value...

Reply:
```python
from timeit import timeit
import numpy as np

n_components = 100
n_samples = 10000

U = np.random.random((n_samples, n_samples))
true_signs = np.sign(np.random.random(n_samples) - 0.5)
V = np.multiply(U, true_signs[None, :]).T


def compute_signs_dot():
    VU = np.dot(V[:n_components, :], U[:, :n_components])
    signs = np.sign(np.diag(VU))
    np.testing.assert_equal(signs, true_signs[:n_components])

def compute_signs_mul():
    diag_VU = np.multiply(V[:n_components, :].T,
                          U[:, :n_components]).sum(axis=0)
    signs = np.sign(diag_VU)
    np.testing.assert_equal(signs, true_signs[:n_components])


a = timeit(compute_signs_dot, number=100)
print("compute_signs_dot: %ss" % a)
b = timeit(compute_signs_mul, number=100)
print("compute_signs_mul: %ss" % b)
```

yields:

```
compute_signs_dot: 2.1252122s
compute_signs_mul: 0.7688638000000001s
```

so I'll do this